aws/credentials/ec2rolecreds/ec2_role_provider.go

```diff
@@ -176,3 +207,21 @@ func requestCred(client *ec2metadata.EC2Metadata, credsName string) (ec2RoleCred
 
 	return respCreds, nil
 }
+
+func requestInstanceIdentityDocument(client *ec2metadata.EC2Metadata) (EC2InstanceIdentityDocument, error) {
+	resp, err := client.GetDynamicData(path.Join(instanceIdentityPath, "document"))
+	if err != nil {
+		return EC2InstanceIdentityDocument{},
+			awserr.New("EC2RoleRequestError",
+				fmt.Sprint("failed to get EC2 instance identity document"), err)
```

Comment:
Probably can drop the `fmt.Sprintf` since not needed. Here and the error case below.